src/wavelet/mod.rs

```diff
@@ -94,8 +94,8 @@ impl WaveletMatrix {
 
         for (level, data) in data.iter_mut().enumerate() {
             let mut total_zeros = 0;
-            for i in 0..num_elements {
-                if bit_lookup(permutation[i], element_len - level - 1) == 0 {
+            for (i, p) in permutation.iter().enumerate().take(num_elements) {
```

Comment:
is take() required here? is `i` required here?

Reply:
I've now removed take() in this and the case below.